thunder/tests/litgpt_model.py

```diff
@@ -134,20 +134,10 @@ def forward(self, input_pos: torch.Tensor, k: torch.Tensor, v: torch.Tensor) ->
         # move the buffer to the activation dtype for when AMP is used
         self.k = self.k.to(k.dtype)
         self.v = self.v.to(v.dtype)
+
         # update the cache
-        # NOTE: `torch._dynamo.is_compiling` is being deprecated, we should update this once all versions have `torch.compiler.is_compiling`.
-        is_compiling = (
-            torch.compiler.is_compiling if hasattr(torch.compiler, "is_compiling") else torch._dynamo.is_compiling
-        )
-        if is_compiling():
-            # inductor doesn't support `index_add` with bfloat16
-            k = self.k.index_copy_(2, input_pos, k)
-            v = self.v.index_copy_(2, input_pos, v)
-            return k, v
-        # See issue: "Support more indexing operators (index_copy and index_add)"
-        k = self.k = torch.index_add(self.k, 2, input_pos, k)
-        v = self.v = torch.index_add(self.v, 2, input_pos, v)
-        # THUNDER bug: cannot return self.k, self.v here (may be cuda graphs related - no minimum repro)
+        k = self.k.index_copy_(2, input_pos, k)
+        v = self.v.index_copy_(2, input_pos, v)
```

Comment:
If this patch works then I don't think we need `OverridenKVCache` at all and can directly rely on `KVCache` from litgpt. Wdyt?

https://github.com/Lightning-AI/litgpt/blob/97ce737b27eababfabd7e94286373df9901df94f/litgpt/model.py#L382-L406

Reply:
Yes, this is the goal. I will follow your suggestion upon green CIs.

Reply:
While it is not apparent from the tests, if you run llama 3 with this, it will not work (out of place `index_copy` will produce sensical results, in-place `index_copy_` not). I have not yet found out why, but as we intend to do benchmarking, this would be interesting to know.

Reply:
(this is from the branch with tom/ggmlquant from PR #754 )

Reply:
Hmm, it breaks the test, simply removing `OverridenKVCache`, that is...

Reply:
The tests are still failing, on the GPU...

Reply:
I would recommend to do it in two/tree steps:
```
k = self.k = torch.index_copy(self.k, 2, input_pos, k)
v = self.v = torch.index_copy(self.v, 2, input_pos, v)
```
should work
```
k = self.k = torch.index_copy_(self.k, 2, input_pos, k)
v = self.v = torch.index_copy_(self.v, 2, input_pos, v)
```
breaks / needs to be fixed
Ultimate goal:
```
k = torch.index_copy_(self.k, 2, input_pos, k)
v = torch.index_copy_(self.v, 2, input_pos, v)
```

Reply:
I have update the PR to include a potential fix...

Reply:
OK, some GPU CIs are green now! :)